Features/35424 tabs component

HiirenP opened the pull request.

dheeraja00 commented on src/components/Tab/Tab.tsx:
```diff
@@ -0,0 +1,35 @@
+  activeTabId?: string;
+  // Set theme for tab
+  theme?: any;
+  children?: any;
```
@HiirenP No need to define children prop as its a pre-defined prop which comes with every component, so we don't have to define its type

HiirenP replied:
resolved!

dheeraja00 commented on src/components/Tab/TabPanel.tsx:
```diff
@@ -0,0 +1,121 @@
+// All prototypes type
+export interface Props {
+  // Children are the tabs and it's content
+  children?: any;
```
@HiirenP Check my above comment

HiirenP replied:
resolved!

dheeraja00 commented on src/components/Tab/TabPanel.tsx:
```diff
@@ -0,0 +1,121 @@
+  // Alignment prop defines the tab's Alignment in tab panel
+  alignment: Alignment;
+  // For Default active tab
+  defaulttabid?: string;
```
@HiirenP Use camel case for all the keys, so it should be `defaultTabId`

HiirenP replied:
I was forced to do this change for only prop because i was getting a warning message something like this. "React does not recognize the `defaultTabId` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `defaulttabid` instead".

spent some time to research on this but not found any solution can you please help me in this?

dheeraja00 replied:
This is a weird issue, I will have a look

dheeraja00 commented on src/components/Tab/TabPanel.tsx:
```diff
@@ -0,0 +1,121 @@
+    return React.Children.map(this.props.children, (child: React.ReactElement<any>, index) => {
+      return React.cloneElement(child, {
+        activeTabId,
+        onClick: this.onTabClick.bind(this, this.props.children[index].props.tabId),
```
@HiirenP 
1. Using `bind` can cause the performance issue, so declare `onTabClick` as arrow function this way you don't have to use bind
2. You don't have to use like `this.props.children[index].props.tabId` I believe this can be changed to `child.props.tabId`

HiirenP replied:
resolved!

dheeraja00 commented:
@HiirenP if you have taken care of all the comments then please try to resolve travis build

HiirenP commented:
Regarding Style/Class props i have added it into tab component earlier, but after confirming the requirement, i had removed it, but anyways i have added it again.
